.flake8

```diff
@@ -0,0 +1,9 @@
+[flake8]
+# https://ljvmiranda921.github.io/notebook/2018/06/21/precommits-using-black-and-flake8/
+ignore = E203, E266, E501, W503, F403, F401 , E402
+max-line-length = 79
+max-complexity = 15
+select = B,C,E,F,W,T4,B9
+exclude =
+ ./src/biokbase/,
```

Comment:
excluding `src/biokbase` will exclude all of the Python code from flake8. Is that the goal, and let Black have at it instead? Or is that just the root directory?

Reply:
@briehl Good catch I meant to exclude the clients

Reply:
I have a few more files to fix now
```
./src/biokbase/workspace/client.py:703:27: W605 invalid escape sequence '\d'
./src/biokbase/narrative/app_util.py:163:1: C901 'app_param' is too complex (16)
./src/biokbase/narrative/app_util.py:364:1: C901 'validate_parameters' is too complex (20)
./src/biokbase/narrative/app_util.py:537:1: C901 'validate_param_value' is too complex (33)
./src/biokbase/narrative/app_util.py:767:1: C901 'transform_param_value' is too complex (19)
./src/biokbase/narrative/viewers.py:29:11: F405 'Network' may be undefined, or defined from star imports: clustergrammer_widget
./src/biokbase/narrative/viewers.py:29:19: F405 'clustergrammer_widget' may be undefined, or defined from star imports: clustergrammer_widget
./src/biokbase/narrative/widgetmanager.py:86:5: C901 'WidgetManager.load_widget_info' is too complex (34)
./src/biokbase/narrative/widgetmanager.py:117:9: F723 syntax error in type comment '(string, int, float, boolean, etc. list? hash?)'
./src/biokbase/narrative/widgetmanager.py:422:5: C901 'WidgetManager.infer_upas' is too complex (23)
./src/biokbase/narrative/contents/narrativeio.py:159:5: C901 'KBaseWSManagerMixin.write_narrative' is too complex (26)
./src/biokbase/narrative/contents/narrativeio.py:317:5: C901 'KBaseWSManagerMixin._process_cell_usage' is too complex (17)
./src/biokbase/narrative/contents/updater.py:82:1: C901 'update_method_cell' is too complex (19)
./src/biokbase/narrative/contents/updater.py:125:9: F841 local variable 'fsm_state' is assigned to but never used
./src/biokbase/narrative/contents/kbasewsmanager.py:464:36: F821 undefined name 'new_notebook'
./src/biokbase/narrative/contents/kbasewsmanager.py:502:20: F821 undefined name 'copy_pat'
./src/biokbase/narrative/contents/kbasewsmanager.py:568:24: F821 undefined name 'fnmatch'
./src/biokbase/narrative/tests/test_appmanager.py:244:5: F811 redefinition of unused 'test_run_app_missing_inputs' from line 165
./src/biokbase/narrative/tests/test_logging.py:74:26: F821 undefined name 'data'
./src/biokbase/narrative/tests/test_narrativeio.py:558:51: F841 local variable 'err' is assigned to but never used
./src/biokbase/narrative/tests/test_jobcomm.py:225:23: F541 f-string is missing placeholders
./src/biokbase/narrative/tests/test_jobcomm.py:281:23: F541 f-string is missing placeholders
./src/biokbase/narrative/common/log_proxy.py:177:16: E713 test for membership should be 'not in'
./src/biokbase/narrative/common/log_proxy.py:238:12: E713 test for membership should be 'not in'
./src/biokbase/narrative/jobs/appmanager.py:697:66: F821 undefined name 'kwargs'
./src/biokbase/narrative/jobs/appmanager.py:833:5: C901 'AppManager._map_inputs' is too complex (16)
./src/biokbase/narrative/jobs/jobmanager.py:228:13: F601 dictionary key 'cell_id' repeated with different values
./src/biokbase/narrative/jobs/jobmanager.py:229:13: F601 dictionary key 'run_id' repeated with different values
./src/biokbase/narrative/jobs/jobmanager.py:231:13: F601 dictionary key 'cell_id' repeated with different values
./src/biokbase/narrative/jobs/jobmanager.py:232:13: F601 dictionary key 'run_id' repeated with different values
./src/biokbase/user_profile/client.py:72:19: E999 SyntaxError: invalid syntax
./src/scripts/fix_ws_metadata/fix_workspace_info.py:61:1: C901 '_fix_single_workspace_info' is too complex (21)
```

Reply:
To save effort, it might make sense to break those down into a series of PRs that we can all work on. Unless Black can automate some of it?

Reply:
Unfortunately it can't be automated.. Though I went through them all and fixed them. There were some orphaned functions, some files that were broken with undefined variables, and some weird stuff going in some of those.  Old files though, from 2013. 
I think these last commits will fix up these issues, waiting for travis to build.